python/mxnet/gluon/data/text.py

```diff
@@ -0,0 +1,160 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# coding: utf-8
+# pylint: disable=
+"""Text datasets."""
+__all__ = ['WikiText2', 'WikiText103']
+
+import io
+import os
+import zipfile
+import shutil
+import numpy as np
+
+from . import dataset
+from ..utils import download, check_sha1
+from ...contrib import text
+from ... import nd
+
+
+class WikiText2(dataset._DownloadedDataset):
+    """WikiText-2 word-level dataset for language modeling, from Salesforce research.
+
+    From
+    https://einstein.ai/research/the-wikitext-long-term-dependency-language-modeling-dataset
+
+    License: Creative Commons Attribution-ShareAlike
+
+    Each sample is a vector of length equal to the specified sequence length.
+    At the end of each sentence, an end-of-sentence token '<eos>' is added.
+
+    Parameters
+    ----------
+    root : str, default '~/.mxnet/datasets/cifar10'
+        Path to temp folder for storing data.
+    segment : str, default 'train'
+        Dataset segment. Options are 'train', 'validation', 'test'.
+    indexer : :class:`~mxnet.contrib.text.indexer.TokenIndexer`, default None
+        The indexer to use for indexing the text dataset. If None, a default indexer is created.
+    seq_len : int, default 35
+        The sequence length of each sample, regardless of the sentence boundary.
+    transform : function, default None
```

Comment:
Dataset now has a transform API. Use that instead of adding transform callback to every dataset

Reply:
Sure, I can take a look at that. What about vision? Are the existing transform options dropped? Where can I find relevant discussion?